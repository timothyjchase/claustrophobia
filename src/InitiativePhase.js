import React from 'react'
import { observer } from 'mobx-react'
import { Button, Icon } from 'semantic-ui-react'
import EventMessage from './EventMessage'

const InitiativePhase = ({ game }) => (
  <div>
    {!!game.event &&
      game.event.phase === 'INITIATIVE' && <EventMessage game={game} />}
    <p>Play this phase normally.</p>
    <Button.Group vertical fluid>
      {!game.oilForYourLamp && (
        <Button onClick={() => game.setOilForYourLamp()}>
          <Icon name="tint" /> Oil For Your Lamp
        </Button>
      )}
      <Button primary fluid onClick={() => game.completeInitiativePhase()}>
        <Icon name="play" /> Next Phase
      </Button>
    </Button.Group>
  </div>
)

export default observer(InitiativePhase)
