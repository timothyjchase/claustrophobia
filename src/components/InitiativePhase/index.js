import React from 'react'
import PropTypes from 'prop-types'
import { Button } from 'semantic-ui-react'

const InitiativePhase = ({
  oilForYourLamp,
  useAuraOfBlessing,
  useOilForYourLamp,
  completeInitiativePhase,
}) => (
  <div>
    <p>Play this phase normally.</p>
    <Button.Group vertical fluid>
      <Button
        icon="heart"
        content="Aura of Blessing (-2 Threat)"
        onClick={useAuraOfBlessing}
      />
      {!oilForYourLamp && (
        <Button
          icon="tint"
          content="Oil For Your Lamp (No Events)"
          onClick={useOilForYourLamp}
        />
      )}
      <Button
        primary
        fluid
        icon="play"
        content="Next Phase"
        onClick={completeInitiativePhase}
      />
    </Button.Group>
  </div>
)

InitiativePhase.propTypes = {
  oilForYourLamp: PropTypes.bool.isRequired,
  useAuraOfBlessing: PropTypes.func.isRequired,
  useOilForYourLamp: PropTypes.func.isRequired,
  completeInitiativePhase: PropTypes.func.isRequired,
}

export default InitiativePhase
