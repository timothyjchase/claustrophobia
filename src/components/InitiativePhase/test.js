import React from 'react'
import InitiativePhase from './index'

describe('<InitiativePhase />', () => {
  const commonProps = {
    useAuraOfBlessing: jest.fn(),
    useOilForYourLamp: jest.fn(),
    completeInitiativePhase: jest.fn(),
  }
  describe('with oilForYourLamp = true', () => {
    const component = shallow(
      <InitiativePhase oilForYourLamp {...commonProps} />,
    )

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with oilForYourLamp = false', () => {
    const component = shallow(
      <InitiativePhase oilForYourLamp={false} {...commonProps} />,
    )

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
