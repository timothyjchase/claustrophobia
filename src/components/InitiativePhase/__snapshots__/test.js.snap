// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InitiativePhase /> with oilForYourLamp = false should match snapshot 1`] = `
<div>
  <p>
    Play this phase normally.
  </p>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Aura of Blessing (-2 Threat)"
      icon="heart"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Oil For Your Lamp (No Events)"
      icon="tint"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Next Phase"
      fluid={true}
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<InitiativePhase /> with oilForYourLamp = true should match snapshot 1`] = `
<div>
  <p>
    Play this phase normally.
  </p>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Aura of Blessing (-2 Threat)"
      icon="heart"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Next Phase"
      fluid={true}
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;
